src/libYARP_cv/include/yarp/cv/Cv-inl.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_CV_CV_INL_H
+#define YARP_CV_CV_INL_H
+
+#include <yarp/os/LogStream.h>
+
+namespace yarp {
+namespace cv {
+
+template <typename T>
+struct cvTypeCode;
+
+template<> struct cvTypeCode<yarp::sig::PixelMono> : std::integral_constant<int, CV_8UC1> {};
+template<> struct cvTypeCode<yarp::sig::PixelMono16> : std::integral_constant<int, CV_16UC1> {};
+template<> struct cvTypeCode<yarp::sig::PixelMonoSigned> : std::integral_constant<int, CV_8SC1> {};
+template<> struct cvTypeCode<yarp::sig::PixelFloat> : std::integral_constant<int, CV_32FC1> {};
+template<> struct cvTypeCode<yarp::sig::PixelInt> : std::integral_constant<int, CV_32SC1> {};
+template<> struct cvTypeCode<yarp::sig::PixelBgr> : std::integral_constant<int, CV_8UC3> {};
+template<> struct cvTypeCode<yarp::sig::PixelRgb> : std::integral_constant<int, CV_8UC3> {};
+template<> struct cvTypeCode<yarp::sig::PixelRgbSigned> : std::integral_constant<int, CV_8SC3> {};
+template<> struct cvTypeCode<yarp::sig::PixelRgbFloat> : std::integral_constant<int, CV_32FC3> {};
+template<> struct cvTypeCode<yarp::sig::PixelRgbInt> : std::integral_constant<int, CV_32SC3> {};
+template<> struct cvTypeCode<yarp::sig::PixelHsv> : std::integral_constant<int, CV_8UC3> {};
+template<> struct cvTypeCode<yarp::sig::PixelHsvFloat> : std::integral_constant<int, CV_32FC3> {};
+template<> struct cvTypeCode<yarp::sig::PixelRgba> : std::integral_constant<int, CV_8UC4> {};
+template<> struct cvTypeCode<yarp::sig::PixelBgra> : std::integral_constant<int, CV_8UC4> {};
```

Comment:
Is this mapping between YARP PixelCode and OpenCV pixel codes documented somewhere?

Reply:
I would also add one minor detail.
The convention where OpenCV use `cv` as a prefix to its function is C-style that they are going to change in favour of namespace `cv`. I can't find the discussion about that anymore, but since `cvTypeCode` and `cvCvtCodeFrom` are already in `yarp::cv` namespace, I propose to remove the `cv` prefix to have much cleaner names 😄

Reply:
The naming convention of modern OpenCV is summarized [here](https://docs.opencv.org/3.4.4/df/de5/tutorial_interoperability_with_OpenCV_1.html):

> All the OpenCV related stuff is put into the _cv_ namespace to avoid name conflicts with other libraries data structures and functions. [...]
> Because the functions are already in a namespace there is no need for them to contain the _cv_ prefix in their name. As such all the new C++ compatible functions don't have this and they follow the camel case naming rule. This means the first letter is small (unless it's a name, like _Canny_) and the subsequent words start with a capital letter (like _copyMakeBorder_).